source/agora/node/Registry.d

```diff
@@ -226,6 +232,12 @@ public class NameRegistry: NameRegistryAPI
 
     public override void postValidator (RegistryPayload registry_payload)
     {
+        if (this.zones[ZoneIndex.Validator].type == ZoneType.secondary)
+        {
+            this.zones[ZoneIndex.Validator].redirect_primary.postValidator(registry_payload);
+            return;
+        }
```

Comment:
Any redirect should happen after we've checked the payload is valid (at the very least the signature), otherwise it's a very obvious DDoS vector.